sample-apps/payment-customization-hide/extensions/hide-first-payment/src/main.rs

```diff
@@ -0,0 +1,74 @@
+use serde::Serialize;
+
+mod api;
+use api::*;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // use stderr to print output
+    eprintln!("Hello World");
+
+    // read from stdin and write to stdout
+    let payload: Payload = serde_json::from_reader(std::io::BufReader::new(std::io::stdin()))?;
+    let mut out = std::io::stdout();
+    let mut serializer = serde_json::Serializer::new(&mut out);
+    function(payload)?.serialize(&mut serializer)?;
+    Ok(())
+}
+
+fn function(payload: Payload) -> Result<FunctionResult, Box<dyn std::error::Error>> {
+    let (input, _config) = (payload.input, payload.configuration);
+    return Ok(build_result(input.payment_methods[0].id.clone()));
```

Comment:
Could you avoid cloning here by passing a reference instead of ownership of the ID?

Reply:
I think we need to clone at some point because the `HideOperation.payment_method_id` needs a type of `ID`, not `&ID`

https://github.com/Shopify/function-examples/pull/92/files/e5b234d53e8b15d4a7075cdc6565fecdc9d767e6#diff-84e00df657808afea93408b8a1469df293a0b938ae80257be70b4d620d1887daR45